File: AnkiDroid/src/main/java/com/ichi2/anki/analytics/AnalyticsDialogFragment.java
package com.ichi2.anki.analytics;

import androidx.fragment.app.DialogFragment;

public abstract class AnalyticsDialogFragment extends DialogFragment {
    @Override
    public void onResume() {
        super.onResume();
        UsageAnalytics.sendAnalyticsScreenView(this);
    }
}

File: AnkiDroid/src/main/java/com/ichi2/anki/analytics/AnalyticsDialogFragment.kt
// SPDX-License-Identifier: GPL-3.0-or-later
// SPDX-FileCopyrightText: Copyright (c) 2018 Mike Hardy <mike@mikehardy.net>

package com.ichi2.anki.analytics

import androidx.annotation.LayoutRes
import androidx.fragment.app.DialogFragment
import com.ichi2.anki.common.analytics.Analytics

abstract class AnalyticsDialogFragment(
    @LayoutRes contentLayoutId: Int = 0,
) : DialogFragment(contentLayoutId) {
    override fun onResume() {
        super.onResume()
        Analytics.sendAnalyticsScreenView(this)
    }
}
